Wpt flags h2

arenevier opened the pull request.

Commit:
websockets/cookies: specify wpt_flags=https for wss variants

Right now, wss variants will fail because they are not served over
https. And an insecure page cannot set a secure cookie.

As a solution, we load those variants with wpt_flags=https. It will
force the test to be loaded over https.

006.https.html was a previous way of solving the same issue. We can now
get rid of it, and simply use 006.html with the correct variant set.

arenevier commented:
@Hexcles @ricea

ricea commented on websockets/constants.js:
```diff
@@ -4,10 +4,31 @@ var PORT = "{{ports[ws][0]}}";
 var PORT_H2 = "{{ports[h2][0]}}";
 
+function url_has_variant(variant) {
+  if (typeof URLSearchParams === "function") {
```
On what platform(s) does this check fail?

arenevier replied:
I removed that check.

ricea commented on websockets/handlers/echo-cookie_wsh.py:
```diff
@@ -3,7 +3,7 @@
 
 def web_socket_do_extra_handshake(request):
-    request.ws_cookie = request.headers_in.get('Cookie')
+    request.ws_cookie = request.headers_in.get('Cookie') or request.headers_in.get('cookie')
```
It seems weird that you need to do this. I file https://github.com/GoogleChromeLabs/pywebsocket3/issues/18 but I don't know whether we'll fix it there.

arenevier replied:
in the case of ws over h2, headers_in are of type H2Headers https://github.com/web-platform-tests/wpt/blob/16d86f34f5b91f9588148fe16624f026bb25430d/tools/wptserve/wptserve/server.py#L615

The alternative would be to modify H2Headers to match fields case-insensitively, though I'm worried of breaking other tests.

ricea replied:
I see, thanks. Maybe experiment with changing H2Headers in a separate CL?

ricea replied:
Actually, I think we can get away with only the lower case versions, since if we are doing a case-sensitive comparison, we must be using H2.

arenevier replied:
Good idea. I have uploaded a new patch.

ricea commented on websockets/interfaces/WebSocket/bufferedAmount/bufferedAmount-blob.html:
```diff
@@ -6,6 +6,7 @@
 <script src=../../../constants.js?pipe=sub></script>
 <meta name="variant" content="">
 <meta name="variant" content="?wss">
+<meta name="variant" content="?htt2">
```
Looks wrong?

arenevier replied:
fixed

ricea reviewed (approved):
lgtm, great, thanks!

Commit:
add wpt_flags=h2 variant to tests

All those tests have been checked to pass manually (or, if the wss
variant fails, to fail in the same way)

Some websocket handlers needed to be modified to accept lower case
headers.

We also need to fix detection of h2 in constants.js: h2 is not specified
with h2 parameter, but with wpt_flags=h2 parameter.

arenevier commented:
@jdm @zqzhang : Can you take a look?
